stride/problem/problem.py

```diff
@@ -111,6 +112,34 @@ def load(self, *args, **kwargs):
                 if getattr(problem_property, grid_property) is None:
                     setattr(problem_property._grid, grid_property, getattr(self, grid_property))
 
+    def time_resample(self, new_step, new_num=None, **kwargs):
+        '''
+        In-place operatin to resample the wavelets and data into a grid with new
+        time-spacing. Sinc interpolation is used.
+
+        Parameters
+        ----------
+        new_step : float
+            The time spacing for the interpolated grid
+        new_num : int
+            The number of time-points, default is calculated to match input pulse
+            length in [s]
+
+        Returns
+        -------
+        '''
+
+        old_step = self.time.step
+        old_num = self.time.num
+        self.grid.time.resample(new_step=new_step, new_num=new_num)
+
+        for shot_id, _ in self.acquisitions._shots.items():
```

Comment:
```
for shot in self.acquisitions.shots:
    shot.wavelets = shot.wavelets._resample(...)
    shot.observed = shot.observed._resample(...)
```

Reply:
This generates the following error

AttributeError: 'int' object has no attribute 'wavelets'